src/app/views/login/SignIn.jsx

```diff
@@ -0,0 +1,342 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import { push } from "react-router-redux";
+import PropTypes from "prop-types";
+
+import LoginForm from "./SignInForm";
+import notifications from "../../utilities/notifications";
+
+import { errCodes } from "../../utilities/errorCodes";
+import user from "../../utilities/api-clients/user";
+import redirectToMainScreen from "../../utilities/redirectToMainScreen";
+import log from "../../utilities/logging/log";
+import ChangePasswordController from "../new-password/changePasswordController";
+import ChangePasswordConfirmed from "../new-password/changePasswordConfirmed";
+import { status } from "../../constants/Authentication";
+
+const propTypes = {
+    dispatch: PropTypes.func.isRequired,
+    isAuthenticated: PropTypes.bool.isRequired,
+    rootPath: PropTypes.string.isRequired,
+    location: PropTypes.object,
+};
+
+export class LoginController extends Component {
+    validationErrors = {};
+    emailErrorMsg = "";
+    passwordErrorMsg = "";
+    session = "";
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            passwordValue: "",
+            emailValue: "",
+            passwordType: "password",
+            status: status.WAITING_USER_INITIAL_CREDS,
+            firstTimeSignIn: false
+        };
+    }
+
+    componentWillMount() {
+        if (this.props.isAuthenticated) {
+            this.props.dispatch(push(`${this.props.rootPath}/collections`));
+        }
+    }
+
+    requestSignIn = credentials => {
+        user.signIn(credentials)
+            .then(response => {
+                let newPasswordRequired = false;
+                // Convert new_password_required string to bool
+                if (response.body != null && response.body.new_password_required !== null) {
```

Comment:
Not sure why you've used a not equal and then a strict not equal operand. Any reason why both checks can't be a strict operand?

Reply:
Ah good spot, so this is the rare case where I actually prefer to use `!= null` rather than the stricter `!== null` as `!= null` catches both `undefied` and `null` whereas `!== null` only captures `null`e.g.
`null != null` (false)
`undefined != null` (false)
`null !== null` (false)
`undefiend !== null` (true)

Will change both to `!= null`